src/foxlator_lib/video.py

```diff
@@ -9,12 +11,24 @@ class VideoError(BaseError):
 
 
 class Video(object):
-    def __init__(self, path: str, fs=FileSystem()):
+    known_extensions = ['mp4']
+
+    def __init__(self, path: str, fs: FileSystem = FileSystem()):
         self.path = path
         self.fs = fs
 
+        self._validate_extension(path)
+
         if (not self.fs.is_file(path)):
-            raise VideoError()
+            raise VideoError(f"Video file: '{path}' doesn't exist")
 
     def get_audio_path(self) -> AudioPath:
         return AudioPath()
+
+    def _validate_extension(self, path: str):
+        extension: str = pathlib.Path(path).suffix
+        if len(extension) == 0:
+            raise VideoError(
+                "Provided file path is invalid or doesn't contain extension")
+        if extension not in self.known_extensions:
+            raise VideoError(f"Unsupported file extension: {extension}")
```

Comment:
The "extention" variable will contain values with a dot prefix, such as ".mp4". For this reason, this method will always return an error. There is onlo small nit from my side. I don't think we need a separate error for the "wrong" extension, we can simply return one message, as in the example above (with the correction)

```
    def _validate_extension(self, path: str):
        extension: str = pathlib.Path(path).suffix
        extension = extension.split(".")[1] if "." in extension else extension
        if extension not in self.known_extensions or not extension:
            raise VideoError(f"Wrong file or extension: {path}")
```

Or just add "." to values in known_extensions

Reply:
Good catch. I'll fix that. For the second part, I would keep it as those are separated cases - lack of extension is not the same as providing extension that is not supported.

Reply:
Corrected.